### PR TITLE
Dwarvish Bloodaxe - add level 3 Berserker advancement

### PR DESCRIPTION
**Pull request by Dalas121**
Dwarvish singleplayer campaigns suffer from issues with variety. Campaigns frequently disallow Outlaws, Gryphons, and Ulfs, leaving only the basic Fighter/Guardsman/Scout/Thunderer available to the player. Without the elements that make the multiplayer Knalgan faction interesting and dynamic, campaign gameplay can suffer.

Ulfs in particular simply aren't useful in many campaigns. Campaigns rely on mulching through large numbers of enemies without losing your own veterans - completely different from the Ulf's role in multiplayer. Even if the player makes the effort to level up an Ulf, there's little reason to recall a Berserker, as berserk makes them hard to keep alive and steals XP from other units.

To help address this issue, I'd like to add a level 3 "Bloodaxe" advancement to the Berserker. Having a L3 advancement makes Berserker XP less wasted, and including drains helps to solve the campaign survivability issue - a skilled player's Bloodaxes can always be at full hp. 8-4 berserk drains is VERY strong, but I think it's a worthy reward for keeping your Ulf alive to L3 - and it only takes a single unlucky 1% chance for your veteran to die.

@doofus-01 thoughts on the art? ~~Sprites from the War of the Gods add-on by ChaosRider, with modifications by Dalas.~~ Sprites by Turin.

https://forums.wesnoth.org/viewtopic.php?p=695051

.

_"According to legend, the dwarvish bloodaxes were forged at the dawn of time in the great stone-fires of Dhur Darihm, when all Irdya was still drowned in flame and soot. It is with them that the first dwarves made war against the troll-mother, laying low her people and forevermore claiming dominion in the underearth._

_While the truth of this legend is a matter of debate, battlefield reports leave no doubt that such weapons exist, and that they confer great resilience upon their wielder. The dwarves’ few remaining bloodaxes are jealously guarded, passed down from father to son and reclaimed from the fallen even at great cost."_

### CODE REVIEW

**Comment by ProditorMagnus** [comment]
Drain looks too strong. [heal_on_hit] would support weaker healing from same events.

**Comment by knyghtmare** [comment]
> At some point, I'd also like to let the player recruit Ulfs in SoF and THoT. I could test that out just myself merging this PR, or we could add Ulfs/Bloodaxes to SoF/THoT right away and give players a chance to see how they perform in the 1.19 development environment. What do you think?

We know core has more stringent requirements for a unit get into. This unit has good art already, but needs to be proven and utilized and proven in mainline first. Thus, I suggest we actually roll the Dwarven Bloodaxe in the dwarven campaigns first before making the core addition: this would include: SoF, THoT, and NR

**Comment by Mirion147** [comment]
I definitely like the concept, giving a strong level 3 reward unit is a great way to develop the berserker line, making this an optional macro level up would definitely be the right way to go imo. I don't think 8-4 is ridiculous, given the strength of other units with drain and the investment required, this seems pretty on par.

I don't think this solves the lack of interesting unit progression that the dwarves struggle with, but that conversing is outside this PR. I'm thinking one of the reasons that campaigns avoid berserkers is because it's hard to really fit them in thematically. They usually become a one off special unit. It's hard to justify an entire army of suicidal units that are completely replaceable with more suicidal units.

**Comment by Hejnewar** [comment]
I dont like the name. 

Also ye it will be very strong. 

> I don't think this solves the lack of interesting unit progression that the dwarves struggle with, but that conversing is outside this PR.

I mean it tries to solve not using ulfs in campaigns. Tbh, I blame that on lack of proper dwarf vs ud. But ye I would like to see more lvl ups for Dwarfs as well.

**Comment by rodolphecombe** [comment]
What about having a lvl 3 with increased physical resistances to improve its survivability? It could be explained by is battle-hardened skin.  

If we go with a drain attack, I  think a secondary attack with less damage would indeed be better. 
For example 9x4 + 4x4 or 5x4 drain

**Comment by Dalas121** [comment]
> We know core has more stringent requirements for a unit get into. This unit has good art already, but needs to be proven and utilized and proven in mainline first. Thus, I suggest we actually roll the Dwarven Bloodaxe in the dwarven campaigns first before making the core addition: this would include: SoF, THoT, and NR

Ok, makes sense to me. I'll change this PR.

To avoid duplicating images/files, I've only added the bloodaxe to THoT (the most "vanilla" dwarf campaign out of the 3).  When I/you/others have time, we can play through and see how he feels.

**Comment by Dalas121** [comment]
> I dont like the name.

I'm open to suggestions, if you or others have any.  I'd prefer to keep the unit named after his weapon, as I liked the flavor of that, but that's just my preference.

> If we go with a drain attack, I think a secondary attack with less damage would indeed be better.
> For example 9x4 + 4x4 or 5x4 drain

I went and tested this out a couple times, but unfortunately the low drains ended up feeling identical (but weaker) to his regular higher-damage attack. Because even the weakest of enemies (excluding the Dark Adept) can out-damage a 4x4 drains, the Bloodaxe wasn't able to heal himself by targeting weak enemies.

**Comment by Dalas121** [comment]
> I dont like the name.

I'm open to suggestions, if you or others have any.  I'd prefer to keep the unit named after his weapon, as I liked the flavor of that, but that's just my preference.

> If we go with a drain attack, I think a secondary attack with less damage would indeed be better.
> For example 9x4 + 4x4 or 5x4 drain

I went and tested this out a couple times, but unfortunately the low drains ended up feeling identical (but weaker) to his regular higher-damage attack. Because even the weakest of enemies (excluding the Dark Adept) can out-damage a 4x4 drains, the Bloodaxe wasn't able to heal himself by targeting weak enemies.

> What about having a lvl 3 with increased physical resistances to improve its survivability? It could be explained by is battle-hardened skin.

On berserk units, I feel that hp/resistances/damage all basically do the exact same thing. Because combat persists until one unit dies, all that matters is the berserk's damage+durability compared to the enemy's damage+durability. Thus, increasing his resistances from 20% to 30% (for example) is basically identical to increasing his damage from 8 to 9 (when fighting in melee, at least).

Even though Drains runs into the same issue (of being identical to damage/hp) when fighting powerful enemies, it can uniquely allow the Bloodaxe (or whatever his name) to actually gain net health when feeble units like mages or archers - that's why I'd thought it could make for particularly interesting gameplay.

**Comment by ForestDragon-wesnoth** [comment]
I agree with Dalas that it's better to keep one attack in some form than to have a secondary attack that is only able to get net-positive healing in very rare edge cases

**Comment by doofus-01** [comment]
> @doofus-01 thoughts on the art? Sprites from the War of the Gods add-on by ChaosRider, with modifications by Dalas.

On the one hand, it fits the unit line yet is easy to distinguish from the ulf/berserker.  On the other hand, this is mostly achieved by giving it huge hair.  Surely something better can be done, once the theme/mechanism is figured out.
ulfserkers-BA

(BTW, I'd be careful with the sprite attribution.  It's possible that is really where it came from, but there's a good chance it was taken from somewhere else before that.)

> > Drain

>I kinda dislike it as it was prevoiusly either very thematically related (bats) or just undead mechanic.

I agree

**Comment by knyghtmare** [comment]
> thoughts on the art? Sprites from the War of the Gods add-on by ChaosRider, with modifications by Dalas.

As @doofus-01 has mentioned the art credit assignment is incorrect. The original author might have been Turin if we visit an old thread from a decade ago: [Here](https://forums.wesnoth.org/viewtopic.php?t=38417)

And woah, I proposed this like a decade ago...how time flies XD

**Comment by Hejnewar** [comment]
Also I just noticed that level 2 has less hair than level 1, and the level 3 would have the perfect hairline. I find that funny.

**Comment by Dalas121** [comment]
> As @doofus-01 has mentioned the art credit assignment is incorrect. The original author might have been Turin if we visit an old thread from a decade ago: Here

Thanks to you both for finding the original (I hope) author! I'll edit the original post.

.

With regards to drain and the overall unit design - I think we need to stay cognizant of the big gulf between player units and AI units:
- Player-controlled lvl3s are irreplaceable, and will often fight many low-level enemies.
- AI-controlled lvl3s are expendable, and will often fight fewer high-level opponents.

Because of this, berserk units behave very different for the player vs an AI:
- Player-controlled berserks are hard to keep alive, which makes veterans berserks much less valuable.
- AI-controlled lvl3 berserks are good at killing player veterans, which makes AI berserks much more valuable.

Therefore, if we only adjust the damage/hp/resists/mp/etc, I'm pretty confident we'll end up with Bloodaxe who's either useless for the player, overpowered for the AI, or both at the same time.

That's why I feel drain is an important addition:
- Player-controlled berserk+drain is good at killing many weak opponents, making it more powerful vs the AI's hordes.
- AI-controlled berserk+drain is poor at killing a single strong opponent, making it less powerful vs the player's veterans.

TLDR: I feel that berserk is UP vs AI, OP vs player. Drain helps balance things out.

**Comment by rodolphecombe** [comment]
Two stupid ideas I got during the night related to the original runic axe idea:
- alternative type: blade/fire damage. The blade had been infused with the warmth of the forge. Good against undead and armored units. Probably ok to balance
- magical special: "You thought you were safe on your forest/village/castle? WRONG! Chop, chop, here comes the Bloodaxe!" (Btw I think the name is fine if we go for runic axe). Probably hard to balance

**Comment by Hejnewar** [comment]
> AI-controlled berserk+drain is poor at killing a single strong opponent (like the player's veterans), compensating for the strength of AI-controlled berserk. 

I dont actually consider this to be like super true, at least looking at this in game if you dont have blade resistance even as melee unit in neutral you will often see 40%+ kill chances.

**Comment by Dalas121** [comment]
> I dont actually consider this to be like super true, at least looking at this in game if you dont have blade resistance even as melee unit in neutral you will often see 40%+ kill chances.

From checking in-game, replacing +4 drains with +4 damage increases his kill chance against a Warlord but decreases it against a Grunt. Seems true to me.

That said, I agree the current stats are really strong - the Bloodaxe goes 35% vs a Warlord, compared to the Ulf's 15% vs a Grunt (healthy+quick vs resilient+quick) - but as a player I'd consider even a 10% chance of death too high to risk a precious level-3. If he does need a nerf, reducing damage down to 7-4 would reduce his Warlord CTK to 15%; identical to the Ulf's CTK vs a Grunt. 


But changing the topic: sine_nomine on the forums has said that he considers the Ulf and Berserker very useful in SP, and uses them regularly. I disagree - but if a lot of players find the Ulf and Berserker useful in a campaign setting even alongside Lords, that would remove the need for this PR entirely.

SP players, how many of you use the Ulf and Berserker regularly in campaigns?

**Comment by knyghtmare** [comment]
> > I dont actually consider this to be like super true, at least looking at this in game if you dont have blade resistance even as melee unit in neutral you will often see 40%+ kill chances.
> 
> From checking in-game, replacing +4 drains with +4 damage increases his kill chance against a Warlord but decreases it against a Grunt. Seems true to me.
> 
> That said, I agree the current stats are really strong - on flat the Bloodaxe goes 35% vs a Warlord, compared to the Ulf's 15% vs a Grunt (healthy+quick vs resilient+quick) - but as a player I'd consider even a 10% chance of death too high to risk a precious level-3. If he does need a nerf, 7-4 would reduce his Warlord CTK to 15%; identical to the Ulf's CTK vs a Grunt.
> 
> . But changing the topic: sine_nomine on the forums has said that he considers the Ulf and Berserker very useful in SP, and uses them regularly. I disagree - but if a lot of players find the Ulf and Berserker useful in a campaign setting even alongside Lords, that would remove the need for this PR entirely.
> 
> SP players, how many of you use the Ulf and Berserker regularly in campaigns?

I do not have access to ulf/ber in mainline campaigns, apart from NR. and, NR is just a nexus of RNG fecal matter which can be passed with just spamming Dwarf Lords, with some Hunstmen as ranged support. Thus, I do not consider their replay-less (lacking evidence) to be of significance. Thus, my stance is:

Let there be a level 3 advancement to Dwarvish Berserker and let it be bind by the `ENABLE_DWARVEN_FANATIC` macro define for the campaign to ensure UMC is not affected.

File: data/core/units/dwarves/Berserker.cfg [hunk 1]
```diff
@@ -9,10 +9,10 @@
     movement_type=dwarvishfoot
     movement=5
     experience=100
+    {AMLA_DEFAULT}
     level=2
     alignment=neutral
-    advances_to=null
-    {AMLA_DEFAULT}
+    advances_to=Dwarvish Bloodaxe
```
Comment (knyghtmare) [inline data/core/units/dwarves/Berserker.cfg:15]: Hey @Dalas121, the Dwarf Bloodaxe is inside the THoT campaign, but this is modifying the core Dwarven Berserker unit. There might be unit type not found error, if used anywhere outside THoT

**Comment by Hejnewar** [comment]
> I do not have access to ulf/ber in mainline campaigns, apart from NR.

There si another place tho, survivals. And I absolutely use ulfs there. Not exactly the same conditions but PvE content as well. In any mission where you have to kill x enemies before next wave they would be useful.

File: data/core/units/dwarves/Berserker.cfg (hunk 1 above)
Comment (Dalas121) [reply data/core/units/dwarves/Berserker.cfg]: Oops, fixed, thanks!

**Comment by Dalas121** [comment]
>  do not have access to ulf/ber in mainline campaigns, apart from NR.

Yeah, I mostly meant UMC campaigns (or things like the survivals that Hej mentioned). If Ulfs/Berserkers are already useful even in a SP/campaign setting, then all that needs to chance for mainline is for them to be enabled in SoF and THoT - with testing, of course.

**Comment by Dalas121** [comment]
Going to close this - seems like it'll be infeasible to add a level 3. Instead, I'm going to try and encourage the use of Ulfserkers/Berserkers by using more ranged enemies in SoF/THoT, as well as adding in the AMLAs from #9635.
